src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerContext.cs

```diff
@@ -53,17 +69,37 @@ public ODataSerializerContext(ResourceContext resource, SelectExpandClause selec
             SkipExpensiveAvailabilityChecks = context.SkipExpensiveAvailabilityChecks;
             MetadataLevel = context.MetadataLevel;
             Items = context.Items;
-
             ExpandedResource = resource; // parent resource
             SelectExpandClause = selectExpandClause;
+            ExpandedNavigationSelectItems = expandedItems;
             EdmProperty = edmProperty; // should be nested property
 
             if (context.NavigationSource != null)
             {
                 IEdmNavigationProperty navigationProperty = edmProperty as IEdmNavigationProperty;
                 if (navigationProperty != null)
                 {
-                    NavigationSource = context.NavigationSource.FindNavigationTarget(NavigationProperty);
+                    bool isDeclaringTypeEntity = navigationProperty.DeclaringType is IEdmEntityType;
+                    if (!isDeclaringTypeEntity)
+                    {
+                        IEnumerable<IEdmNavigationPropertyBinding> bindings = context.NavigationSource.FindNavigationPropertyBindings(navigationProperty);
+                        foreach (IEdmNavigationPropertyBinding binding in bindings)
+                        {
+                            if (binding.NavigationProperty == navigationProperty)
```

Comment:
>if (binding.NavigationProperty == navigationProperty) [](start = 28, length = 53)

It's not correct, right?

For example:

ComplexType
{
      EntityType NavigationProperty {get;set;}
}

EntityType
{
       ComplexType complexProperty1 {get;set;}
       ComplexType complexProperty2 {get;set;}
}

<EntitySet Name="Set" EntityType="NS.EntityType" >
     <NPB Path="complexProperty1/NavigationProperty" Target="SetA" />
     <NPB Path="complexProperty2/NavigationProperty" Target="SetB" />
</EntitySet>


So, for the same navigation property, only "SetA" is returned, and "SetB" never be returned?